listener/udp.go

```diff
@@ -0,0 +1,109 @@
+// Package listener handles the ingestion for polymur
+package listener
+
+import (
+	"log"
+	"net"
+	"time"
+
+	"github.com/chrissnell/polymur/statstracker"
+)
+
+// UDPListenerConfig holds configuration for the UDP listener
+type UDPListenerConfig struct {
+	IncomingQueue chan []*string
+	FlushTimeout  int
+	FlushSize     int
+	Stats         *statstracker.Stats
+	IP            string
+	Port          int
+	Zone          string
+}
+
+// UDPListener Listens for messages.
+func UDPListener(config *UDPListenerConfig) {
+	log.Printf("Metrics listener started: [UDP] %s:%d\n", config.IP, config.Port)
+	server, err := net.ListenUDP(
+		"udp",
+		&net.UDPAddr{IP: net.ParseIP(config.IP), Port: config.Port},
+	)
+	if err != nil {
+		log.Fatalf("Listener error: %s\n", err)
+	}
+
+	messages := make(chan string, 128)
+	go udpMessageBatcher(messages, config)
+
+	buf := make([]byte, 1024)
+	go func() {
+		defer server.Close()
+		defer close(messages)
+		for {
+			n, _, err := server.ReadFromUDP(buf)
+			if err != nil {
+				log.Printf("UDP read error: %s\n", err)
+				continue
+			}
+			// We need to have each message in its own
+			// array so that the slices sent to messages don't
+			// have their data changed underneath if a new packet
+			// is received before processing the previous one.
+			msg := make([]byte, n)
+			copy(msg, buf[:n])
+			messages <- string(msg)
```

Comment:
I *think* you can just do this:
```
messages <- string(buf[:n])
```
instead of creating a new `msg` with every packet.  That could put a lot of pressure on the garbage collector.